monitor.go

```diff
@@ -293,7 +331,7 @@ func (monitor *Monitor) Close() error {
 // calling Close on Monitor or encountering a netlink conn error while Receive.
 // Caller may receive a MonitorEventTypeOOB event which contains an error we didn't
 // handle, for now.
-func (cc *Conn) AddMonitor(monitor *Monitor) (chan *MonitorEvent, error) {
+func (cc *Conn) AddMonitor(monitor *Monitor) (chan *MonitorEvents, error) {
```

Comment:
There are existing users of this method, so we should not change its API.

Maybe we could introduce a new AddGenerationalMonitor() method with your suggested signature, and make AddMonitor work with generations, unpack the events, put them on a channel of the old type?

Reply:
AddGenerationalMonitor introduced, thank You